src/Files.App.Storage/NativeStorage/NativeStorageService.cs

```diff
@@ -22,12 +24,31 @@ public Task<IFile> GetFileAsync(string id, CancellationToken cancellationToken =
 		}
 
 		/// <inheritdoc/>
-		public Task<IFolder> GetFolderAsync(string id, CancellationToken cancellationToken = default)
+		public async Task<IFolder> GetFolderAsync(string id, CancellationToken cancellationToken = default)
 		{
 			if (!Directory.Exists(id))
 				throw new DirectoryNotFoundException();
 
-			return Task.FromResult<IFolder>(new NativeFolder(id));
+			// A special folder should use the localized name
+			if (PathHelpers.IsSpecialFolder(id))
+			{
+				var storageFolder = await TryGetStorageFolderAsync(id);
+				return new NativeFolder(id, storageFolder?.DisplayName);
```

Comment:
@gave92 Do you know a shell method to do it without the Storage.* overhead? I've searched the docs but the methods there don't accept a specific culture (in this case it would be the app current culture)